### PR TITLE
Improved block.rst

### PR DESCRIPTION
**Pull request by dgarcia360**
Reorganized rollbacks concept under blocks to prepare for finalization changes.

### CODE REVIEW

File: source/concepts/block.rst [hunk 1]
```diff
@@ -59,6 +48,35 @@ Serialization of the block header.
...
+********
+Creation
+********
+
+|codename| calls the first block in the chain the **nemesis block**.
+The first block is defined before launching a new network and sets the distribution of the network currency mosaics.
+
+The process of creating the subsequent new blocks is called :doc:`harvesting <harvesting>`.
+
+Blocks are created by :doc:`accounts <account>` based on their importance score.
```
Comment (matthewkim93) [inline source/concepts/block.rst:60]: Blocks are created by :doc:`accounts <account>`, which are  chosen by the :doc:`consensus algorithm <consensus-algorithm>`based on their importance score.

File: source/concepts/block.rst [hunk 2]
```diff
@@ -59,6 +48,35 @@ Serialization of the block header.
...
+The :doc:`consensus algorithm <consensus-algorithm>` chooses each round the account that harvest the block.
```
Comment (matthewkim93) [inline source/concepts/block.rst:61]: The consensus algorithm determines a new account to harvest the subsequent block after each block creation.

File: source/concepts/block.rst [hunk 3]
```diff
@@ -59,6 +48,35 @@ Serialization of the block header.
...
+*********
+Rollbacks
+*********
+
+Blockchains are designed in a way that, in the presence of a network failure or partition, the recent blocks might need to be rolled back.
+
+The rewrite limit is the maximum number of blocks that can be rolled back.
+Hence, forks can only be resolved up to a certain depth too.
+
+|codename|'s public network has a rewrite limit of ``398`` blocks, being this limit :ref:`configurable per network <config-network-properties>`.
```
Comment (matthewkim93) [inline source/concepts/block.rst:77]: |codename|'s public network has a rewrite limit of ``398`` blocks, but this limit is :ref:`configurable per network <config-network-properties>`.

File: source/concepts/block.rst [hunk 4]
```diff
@@ -59,6 +48,35 @@ Serialization of the block header.
...
+At this point, the transactions linked to the block are permanently recorded on the blockchain.
```
Comment (matthewkim93) [inline source/concepts/block.rst:78]: The transactions linked to a block are permanently recorded on the blockchain once the number of confirmation blocks (subsequent blocks) surpasses the maximum number of rollback blocks.

File: source/concepts/transaction.rst [hunk 5]
```diff
@@ -153,24 +153,15 @@ If valid, the harvester stores the transaction in a block and reaches the **conf
 Continuing the previous example, the transaction gets processed and the amount stated gets transferred from the signer's account to the recipient's account.
 Additionally, the :doc:`transaction fee <fees>` is deducted from the signer's account.
 
-The transaction has **zero confirmations** at this point.
+The transaction has zero confirmations at this point.
 When another block is added to the blockchain, the transaction has one confirmation.
 The next block added to the chain will give it two confirmations and so on.
 
-.. _rollbacks:
+Under certain circumstances, like network failure or partition, the most recent confirmed blocks need to be reversed.
```
Comment (matthewkim93) [inline source/concepts/transaction.rst:160]: Under certain circumstances, like network failure or partition, the most recent confirmed blocks might need to be reversed.

File: source/concepts/transaction.rst [hunk 6]
```diff
@@ -153,24 +153,15 @@ If valid, the harvester stores the transaction in a block and reaches the **conf
...
+:ref:`Rollbacks <rollbacks>` are characteristic in blockchain systems and essential to resolve the forks of the blockchain.
```
Comment (matthewkim93) [inline source/concepts/transaction.rst:161]: :ref:`Rollbacks <rollbacks>` are characteristic of blockchain systems and essential to resolve the forks of the blockchain.

File: source/concepts/transaction.rst [hunk 7]
```diff
@@ -153,24 +153,15 @@ If valid, the harvester stores the transaction in a block and reaches the **conf
...
-
-Blockchains are designed in a way that, under certain circumstances, recent blocks need to be rolled back.
-These are essential to resolve the forks of the blockchain.
-
-The rewrite limit is the maximum number of blocks that can be rolled back.
-Hence, forks can only be resolved up to a certain depth too.
-
-|codename|'s public network has a rewrite limit of ``398`` blocks, being this limit :ref:`configurable per network <config-network-properties>`.
-Once a transaction has more than ``maxRollBackConfirmations`` value, it cannot be reversed.
+In the public network, a transaction is considered to be irrevocable when it receives ``398`` confirmations, being this value :ref:`configurable per network <config-network-properties>`.
```
Comment (matthewkim93) [inline source/concepts/transaction.rst:163]: In the public network, a transaction is considered to be irrevocable when it receives ``398`` confirmations. But the maximum number of blocks that can be rolled back is :ref:`configurable per network <config-network-properties>`.

File: source/concepts/transaction.rst [hunk 8]
```diff
@@ -153,24 +153,15 @@ If valid, the harvester stores the transaction in a block and reaches the **conf
...
+In other words, it is necessary to wait at least ``398`` blocks after a transaction receives its first confirmation to have the guarantee that it cannot be reversed.
```
Comment (matthewkim93) [inline source/concepts/transaction.rst:164]: In other words, it is necessary to wait at least ``398`` blocks after a transaction receives its first confirmation to guarantee that it cannot be reversed on the public network.

**Comment by dgarcia360** [comment]
@matthewkim93 Thanks for the edits!

**Pull request merged**